File: dvbViewerController/src/main/java/org/dvbviewer/controller/ui/widget/NumericEditTextPreference.java
package org.dvbviewer.controller.ui.widget;

import android.content.Context;
import android.text.InputType;
import android.util.AttributeSet;

import androidx.preference.EditTextPreference;

public class NumericEditTextPreference extends EditTextPreference implements  InputTypePref{

    public NumericEditTextPreference(Context context, AttributeSet attrs, int defStyleAttr,
                                     int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public NumericEditTextPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public NumericEditTextPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public NumericEditTextPreference(Context context) {
        super(context);
    }

    @Override
    public int getInputType() {
        return InputType.TYPE_CLASS_NUMBER;
    }
}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/ui/widget/NumericEditTextPreference.kt
package org.dvbviewer.controller.ui.widget

import android.content.Context
import android.text.InputType
import android.util.AttributeSet

import androidx.preference.EditTextPreference

class NumericEditTextPreference : EditTextPreference, InputTypePref {

    override val inputType: Int
        get() = InputType.TYPE_CLASS_NUMBER

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int,
                defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}
}